Entwurf/Zuarbeit Entwurf.tex

```diff
@@ -0,0 +1,84 @@
+Die Controller kommunizieren außerdem noch mit dem Model. Dieses besitzt als alleinige Komponente den Zugang zur Datenbank. Damit ist auch hier ohne großen Aufwand ein Austausch mit lokalen Daten möglich.
+Durch diese Gestaltung der Komponenten haben wir außerdem eine bessere Wiederverwendbarkeit von Komponenten geschaffen. 
+
+ \subsection{Die Klassendiagramme im Detail}
+ \subsubsection{Das Paket Start}
+ 
+ \begin{figure}
+  \includegraphics[scale=0.5]{Entwurf_KD_Feinentwurf_Start.png}
+  \label{fig:Klassendiagramm_Start}
+  \caption{Feinentwurf des Pakets Start}
+ \end{figure} 
+ 
+ Das Klassendiagramm in Abbildung \ref{fig:Klassendiagramm_Start} zeigt die Klassen der Startansicht. Hier kann ausgewählt werden, welche Rolle der Benutzer einnimmt. 
+ Daher gibt es hier nur eine View und einen Controller. Diese bilden die Rollenauswahl ab.
+ 
+ \subsubsection{Das Paket Student}
+ 
+ \begin{figure}
+\includegraphics[scale=0.25]{Entwurf_KD_Feinentwurf_Student.png}
+\label{fig:Klassendiagramm_Student}
+\caption{Feinentwurf des Pakets Student}
+ \end{figure} 
+
+Das Klassendiagramm in Abbildung \ref{fig:Klassendiagramm_Student} zeigt die Klassen der Studentenansicht. Es ist in drei Controller und Views unterteilt. Eine zur Gruppenauswahl, für die Termine und eine für den Buchungsdialog.
+ Jede View ist wieder mit dem entsprechenden Controller verbunden. Zusätzlich dazu gibt es Verbindungen zum Model um notwendige Daten der Termine und die Gruppen zu bestimmen.
+ 
+ \subsubsection{Das Paket Admin}
+ \begin{figure}
+  \includegraphics[scale=0.25]{Entwurf_KD_Feinentwurf_Admin.png}
+  \label{fig:Klassendiagramm_Admin}
+  \caption{Feinentwurf des Pakets Admin}
+ \end{figure} 
+
+ Die Abbildung \ref{fig:Klassendiagramm_Admin} zeigt die Klassen der Adminansicht. Es gibt vier Controller mit den dazugehörigen Views. Der Controller für die Appointments und für den Bearbeitungsdialog benötigen im Model die Appointments. Der Groupcontroller benötigt Zugriff auf die Gruppen im Model
+ 
+ \subsubsection{Model und Repository}
+ 
+ \begin{figure}
+  \includegraphics[scale=0.20]{Entwurf_KD_Models-Repo.png}
+  \label{fig:Klassendiagramm_Models-Repo}
+\caption{Feinentwurf von Model und das Repository}
+ \end{figure} 
+
+Im Paket Model (siehe Abbildung \ref{fig:Klassendiagramm_Models-Repo}) befinden sich die Klassen Appointment, Reservation, Booking, TimeWindow und Group. Die Klasse Appointment ist für die Termine zuständig. Die Klasse Reservation kümmert sich um Reservierungen. Booking um das Buchen von Terminen, TimeWindow um die Zeitfenster der Termine und Buchungen und die Klasse Group um alle Belange der Gruppen, wie zum Beispiel anlegen oder löschen.
+Im Paket Repository (ebenfalls in Abbildung \ref{fig:Klassendiagramm_Models-Repo}) befindet sich die Klasse DBRepository. Sie kümmert sich um die Verbindung zur Datenbank und gibt zum Beipiel alle Gruppen aus oder fügt Termine ein.
```

Comment:
Hier sollte man noch beschreiben, dass die Zugangsdaten zur Datenbank in der Datei db.properties abgelegt werden. Dadurch kann man leicht eine andere Datenbank verwenden, wie es z.B. bei der Entwicklung mit der Testdatenbank geschieht.